plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -0,0 +1,194 @@
+            return LogUniformDistribution(value.start, value.end)
+        else:
+            if "int" in value.tags:
+                return IntUniformDistribution(int(value.start), int(value.end))
+            return UniformDistribution(value.start, value.end)
+
+    raise NotImplementedError("{} is not supported by Optuna sweeper.".format(override))
+
+
+class OptunaSweeperImpl(Sweeper):
+    def __init__(
+        self, optuna_config: OptunaConfig, search_space: Optional[DictConfig]
+    ) -> None:
+        self.optuna_config = optuna_config
+        self.search_space = {}
+        if search_space:
+            assert isinstance(search_space, DictConfig)
+            self.search_space = {
+                x: create_optuna_distribution_from_config(y)
+                for x, y in search_space.items()
+            }
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+
+        self.config = config
+        self.config_loader = config_loader
+        self.launcher = Plugins.instance().instantiate_launcher(
+            config=config, config_loader=config_loader, task_function=task_function
+        )
+        self.sweep_dir = config.hydra.sweep.dir
+
+    def sweep(self, arguments: List[str]) -> None:
+        assert self.config is not None
+        assert self.launcher is not None
+
+        parser = OverridesParser.create()
+        parsed = parser.parse_overrides(arguments)
+
+        search_space = dict(self.search_space)
+        for override in parsed:
+            search_space[
+                override.get_key_element()
+            ] = create_optuna_distribution_from_override(override)
+
+        if self.optuna_config.sampler:
+            sampler_class = getattr(optuna.samplers, self.optuna_config.sampler)
+        else:
+            sampler_class = TPESampler
+
+        if sampler_class in {CmaEsSampler, RandomSampler, TPESampler}:
+            sampler = sampler_class(seed=self.optuna_config.seed)
+        else:
+            sampler = sampler_class()
+
+        study = optuna.create_study(
+            study_name=self.optuna_config.study_name,
+            storage=self.optuna_config.storage,
+            sampler=sampler,
+            direction=self.optuna_config.direction,
+        )
+
+        batch_size = self.optuna_config.n_jobs
+        n_trials_to_go = self.optuna_config.n_trials
+
+        while n_trials_to_go > 0:
+            batch_size = min(n_trials_to_go, batch_size)
+
+            trials = [study._ask() for _ in range(batch_size)]
+            for trial in trials:
+                for param_name, distribution in search_space.items():
+                    trial._suggest(param_name, distribution)
+
+            overrides = []
+            for trial in trials:
+                overrides.append(
+                    tuple(f"{name}={val}" for name, val in trial.params.items())
+                )
```

Comment:
can these two loops be merged into one?
```suggestion
            overrides = []
            for trial in trials:
                for param_name, distribution in search_space.items():
                    trial._suggest(param_name, distribution)
                
                overrides.append(
                    tuple(f"{name}={val}" for name, val in trial.params.items())
                )
```

Reply:
Thanks! I applied your suggestion.